starlette/middleware/sessions.py

```diff
@@ -49,14 +49,17 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
 
         async def send_wrapper(message: Message) -> None:
             if message["type"] == "http.response.start":
+                # Detect ASGI root path
+                path = scope["root_path"] or "/"
```

Comment:
Note that ]`root_path` is Optional](https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope) so this could fail with a KeyError.

Perhaps we want this?...

```suggestion
                path = scope.get("root_path", "") or "/"
```